### PR TITLE
getOntology tool

### PR DESCRIPTION
**Pull request by emily101-gif**

### CODE REVIEW

**Comment by emily101-gif** [comment]
@pcm32 there are also 2 scripts for this tool - an R and python script. 
Should I add them both to the tool repository and just have the XML file refer to the python script here. Thanks :)

**Comment by pcm32** [comment]
Yes, in one of the tools I noticed that the R script was calling the python one, or the other way around. So add both of them.

File: flowtools/flowcl/getOntology.xml [hunk 1]
```diff
@@ -1,25 +1,19 @@
-<tool id="get_cell_ontology" name="Explore cell ontology" version="1.0">
+<tool id="get_cell_ontology" name="Explore cell ontology" version="1.0+galaxy0">
```
Comment (pcm32) [inline flowtools/flowcl/getOntology.xml]:
```suggestion
<tool id="get_cell_ontology" name="Explore cell ontology" version="1.0+galaxy0" profile="18.01">
```
Please set profiles like this in general.

File: flowtools/flowcl/getOntology.xml [hunk 2]
```diff
@@ -1,25 +1,19 @@
...
   <description>given a phenotype using flowCL</description>
   <requirements>
-    <requirement type="package" version="2.0.2">ig-flowtools</requirement>
+    <requirement type="package" version="2.8">jinja2</requirement>
+    <requirement type="package" version="1.14.0">bioconductor-flowcl</requirement>
   </requirements>
   <stdio>
     <exit_code range="2" level="fatal" description="There was a problem running flowCL. You might want to check your marker names - See stderr for more details." />
```
Comment (pcm32) [inline flowtools/flowcl/getOntology.xml:9]: Please add a range here that covers 1 as well.

**Comment by emily101-gif** [comment]
> Yes, in one of the tools I noticed that the R script was calling the python one, or the other way around. So add both of them.
@pcm32 

Sorry just seen this comment and have now added the R script.

Regarding the range - I was basing the XML script on the immport-galaxy repo so do you think it is still worth editing the ranges?

https://github.com/ImmPortDB/immport-galaxy/blob/8f353d1f9b4e0d044e1a9d0b1f928b440df78b8c/tools/flowtools/analysis/FlowSOMMApIndividualFCS.xml

**Comment by pcm32** [comment]
> Regarding the range - I was basing the XML script on the immport-galaxy repo so do you think it is still worth editing the ranges?

Yes. If there are error codes that escape the range, the script can fail but Galaxy will think that it had a successful execution, which shouldn't be.

**Comment by pcm32** [comment]
This is failing for the same reasons as #75 now (access to URL that is not providing a service).

**Comment by pcm32** [comment]
This is failing due to changed ontology content (which is bound to be repeated in the future):

```

--- local_file
+++ history_data
@@ -8,17 +8,17 @@
                       "PR_000001004 + more"                                                  
 Marker Label          "1) CD4 molecule 2) CD4 molecule 3) CD4 molecule 4) CD4 molecule 5)"   
                       "CD4 molecule + more"                                                  
-Marker Key            "1) { } CD4+ ( ) [ alpha-beta T cell receptor complex+, T cell"        
-                      "receptor co-receptor CD8+ ] 2) { } CD4+ ( ) [ alpha-beta T cell"      
-                      "receptor complex+, T cell receptor co-receptor CD8+ ] 3) { } CD4+ ( )"
-                      "[ alpha-beta T cell receptor complex+, T cell receptor co-receptor"   
-                      "CD8+ ] 4) { } CD4+ ( ) [ T-cell surface glycoprotein CD8 alpha"       
-                      "chain-, alpha-beta T cell receptor complex+ ] 5) { } CD4+ ( ) [ T"    
-                      "cell receptor co-receptor CD8-, alpha-beta T cell receptor complex+," 
-                      "CD3+ ] + more"                                                        
+Marker Key            "1) { } CD4+ ( ) [ T-cell surface glycoprotein CD8 alpha chain-,"      
+                      "alpha-beta T cell receptor complex+ ] 2) { } CD4+ ( ) [ alpha-beta T" 
+                      "cell receptor complex+, T cell receptor co-receptor CD8+ ] 3) { }"    
+                      "CD4+ ( ) [ alpha-beta T cell receptor complex+, T cell receptor"      
+                      "co-receptor CD8+ ] 4) { } CD4+ ( ) [ alpha-beta T cell receptor"      
+                      "complex+, T cell receptor co-receptor CD8+ ] 5) { } CD4+ ( ) [ T cell"
+                      "receptor co-receptor CD8-, alpha-beta T cell receptor complex+, CD3+" 
+                      "] + more"                                                             
 Score (Out of 1)      "1) 0.333 2) 0.333 3) 0.333 4) 0.333 5) 0.25 + more"                   
-Cell ID               "1) CL_0000809 2) CL_0002427 3) CL_0002428 4) CL_0000810 5) CL_0000492"
+Cell ID               "1) CL_0000810 2) CL_0000809 3) CL_0002427 4) CL_0002428 5) CL_0000492"
                       "+ more"                                                               
-Cell Label            "1) double-positive, alpha-beta thymocyte 2) resting double-positive"  
-                      "thymocyte 3) double-positive blast 4) CD4-positive, alpha-beta"       
-                      "thymocyte 5) CD4-positive helper T cell + more"                       
+Cell Label            "1) CD4-positive, alpha-beta thymocyte 2) double-positive, alpha-beta" 
+                      "thymocyte 3) resting double-positive thymocyte 4) double-positive"    
+                      "blast 5) CD4-positive helper T cell + more" 
```

Will leave to be compared as similar size.

**Review by pcm32 — APPROVED** [review]
Checked as well against https://github.com/ImmPortDB/immport-galaxy/blob/master/tools/flowtools/analysis/getOntology.xml, nothing newer there.

**Pull request merged**